client/my-sites/comments/comment-list/index.jsx

```diff
@@ -100,26 +90,19 @@ export class CommentList extends Component {
 		);
 	};
 
-	getTotalPages = () => Math.ceil( this.props.comments.length / COMMENTS_PER_PAGE );
-
-	hasCommentJustMovedBackToCurrentStatus = commentId => this.state.lastUndo === commentId;
+	getTotalPages = () => Math.ceil( this.props.commentsCount / COMMENTS_PER_PAGE );
 
 	isCommentSelected = commentId => !! find( this.state.selectedComments, { commentId } );
 
 	isRequestedPageValid = () => this.getTotalPages() >= this.props.page;
 
-	isSelectedAll = () => {
-		const { page } = this.props;
-		const { selectedComments } = this.state;
-		const visibleComments = this.getCommentsPage( this.getComments(), page );
-		return selectedComments.length && selectedComments.length === visibleComments.length;
-	};
+	isSelectedAll = () =>
+		this.state.selectedComments.length &&
+		this.state.selectedComments.length === this.props.comments.length;
 
-	setSortOrder = order => () => {
-		this.setState( {
-			sortOrder: order,
-			page: 1,
-		} );
+	setOrder = order => () => {
+		this.props.setOrder( order );
+		this.setState( { page: 1 } );
```

Comment:
Page isn't in component state anymore.
This can be removed and we can simply pass `this.props.setOrder` down to the children instead.